### PR TITLE
Ap/feature/dashboard home

### PR DESCRIPTION
**Pull request by Adelle-Pitsas**
**What does this PR do?**
Adds the user dashboard (finally) and their pending requests

**Where should your reviewer start**
-`Dashboard`, `DashboardMenu`, `LibraryBook`, `MyBooks`
- There are also some new routes in `App`

**What (if any) features are you implementing**
- Adding the user dashboard feature, where they should see the books in their library (greyed out if its unavailable)
- Adding pending requests (no actual functionality - WIP)

**Were there any issues that arose?**
- Silly async issues that have been resolved
- Its come to my attention we are all using different components for displaying the same sort of thing (books on a page)--See below

**Is there anything that you need from your teammate?**
- Do we need to consolidate all of our components that display one book?

### CODE REVIEW

File: src/Components/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  pathname: string,
+  search: string,
+  hash: string,
+  state: null,
+  key: string
+}
+
+
+const Dashboard = () => {
+  let location: Location = useLocation()
```
Comment (FrazierLE) [inline src/Components/Dashboard/Dashboard.tsx]: Curious as to what you're using location for

File: src/Components/DashboardMenu/DashboardMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { NavLink, useLocation } from "react-router-dom";
+
+
+const DashboardMenu = () => {
+  return(
+    <div>
+      <NavLink to='/dashboard'><button>My Books</button></NavLink>
+      <NavLink to='/dashboard/my-borrowed-books'><button>My borrowed books</button></NavLink>
+      <NavLink to='/dashboard/pending-requests'><button>Pending Requests</button></NavLink>
```
Comment (FrazierLE) [inline src/Components/DashboardMenu/DashboardMenu.tsx:10]: The dashboard menu disappears when I'm on pending requests

File: src/Components/LibraryBook/LibraryBook.css [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.book-unavailable {
+  opacity: .5;
+}
```
Comment (FrazierLE) [inline src/Components/LibraryBook/LibraryBook.css:3]: 👏

File: src/Components/PendingRequests/PendingRequests.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from "react";
+import { gql, useQuery } from '@apollo/client'
+import Request from "../Request/Request";
+
+const REQUESTS = gql `
+  query user {
+    user(id:1) {
```
Comment (FrazierLE) [inline src/Components/PendingRequests/PendingRequests.tsx:7]: I believe we will need to make this id dynamic. I can show you what we did for book details

File: src/components/App/App.tsx [hunk 5]
```diff
@@ -12,16 +16,43 @@ interface Location {
   state: null,
   key: string
 }
+
+const ALL_USERS = gql `
+  query user {
+    user(id:1) {
```
Comment (FrazierLE) [inline src/components/App/App.tsx]: I think this id will also have to be dynamic

**Review by FrazierLE — COMMENTED** [review]
I added some comments. Looks good!

File: src/Components/Dashboard/Dashboard.tsx (hunk 1 above)
Comment (Adelle-Pitsas) [reply src/Components/Dashboard/Dashboard.tsx]: Yep! Remanent of past iteration of Dashboard.... removing

File: src/Components/DashboardMenu/DashboardMenu.tsx (hunk 2 above)
Comment (Adelle-Pitsas) [reply src/Components/DashboardMenu/DashboardMenu.tsx:10]: Good catch. Adding it to `line 41` in `App.tsx` which is getting to be kind of excessive. Do you know how to conditionally render an element on multiple routes?

File: src/Components/PendingRequests/PendingRequests.tsx (hunk 4 above)
Comment (Adelle-Pitsas) [reply src/Components/PendingRequests/PendingRequests.tsx:7]: We talked about this in slack thread -- can't make this dynamic until we have Josh's contribution of saving the user in app

**Review by JPierce28 — APPROVED** [review]
This looks great, as soon as I have added saving a user to App I will let you know so you can make everything dynamic

**Pull request merged**